synapse/http/server.py

```diff
@@ -551,7 +558,7 @@ def __init__(
         request: Request,
         iterator: Iterator[bytes],
     ):
-        self._request = request
+        self._request = request  # type: Optional[Request]
```

Comment:
We set it to `None` when we stop producing so we need to consider it optional here.